### PR TITLE
Containerd bug fixes

### PR DESCRIPTION
**Pull request by adarsh-zededa**
This PR solve the following containerd issues:
1) Snapshot not deleted on appInstance delete
_Resolution: Added a snapshot delete call in ctrRm() (container remove)_

2) Containerd bundle uuid and appInstance's uudi (in xen config) were not same.
_Resolution: Instead of generating new uuid while creating container's bundle, using the same uuid from domainStatus._

3) EntryPoint was not included as part cmdline file in container bundle
_Resolution: Included Entrypoint in cmdline file_

4) Remove unused variables from mount_disk.sh 
_Resolution:  Remove `$major` and `$minor` variables which are never used._ 

5) environment variables with special characters failed to export
_Resolution: Added quotes around the environment values_

Also added the files generated when we ran `make run-compose`  to .gitignore

### CODE REVIEW

File: pkg/pillar/cmd/domainmgr/containerd.go [hunk 1]
```diff
@@ -127,24 +126,39 @@ func ctrStop(containerID string, force bool) error {
...
 	if err := container.Delete(ctrdCtx); err != nil {
 		log.Errorf("Unable to delete container: %v. %v", containerID, err.Error())
-		return fmt.Errorf("cleanUpContainer: Unable to delete container: %v. %v", containerID, err.Error())
+		if !silent {
+			return fmt.Errorf("cleanUpContainer: Unable to delete container: %v. %v", containerID, err.Error())
+		}
+
+	}
+	snapshotter := ctrdClient.SnapshotService(defaultSnapshotter)
+	if err = snapshotter.Remove(ctrdCtx, getSnapshotName(containerID)); err != nil {
```
Comment (deitch) [inline pkg/pillar/cmd/domainmgr/containerd.go]: Why was this necessary? The garbage collector should see a snapshot that is not used by any parent (i.e. container) and thus delete it.

Comment (adarsh-zededa) [reply pkg/pillar/cmd/domainmgr/containerd.go]: Yeah, that was the assumption, but I noticed the snapshot were still present even after calling `container.Delete()` so added a fix where I'm explicitly deleting the snapshot.

Comment (deitch) [reply pkg/pillar/cmd/domainmgr/containerd.go]: I have been digging into how the GC works on containerd, so I will see if anything comes up. I am guessing it cannot hurt to delete it, but what happens if it is GC-ed and you try to delete it? Do we get an error?

Also, can you try to delete container without deleting snapshot, and seeing what labels are on the snapshot? Labels affect GC.

File: pkg/pillar/cmd/domainmgr/containerd.go [hunk 2]
```diff
@@ -127,24 +126,39 @@ func ctrStop(containerID string, force bool) error {
...
+func ctrRm(containerID string, silent bool) error {
 	log.Infof("ctrRm %s\n", containerID)
 
 	container, err := loadContainer(containerID)
 	if err != nil {
-		return err
+		log.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		if !silent {
+			return fmt.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		}
```
Comment (eriknordmark) [inline pkg/pillar/cmd/domainmgr/containerd.go]: Can avoid duplication in all these cases if you do
 err = fmt.Errorf(...)
 log.Error(err)
 if !silent {
    return err
}


Also, in the silent case shouldn't you return nil? How do we know it is safe to drive on when a previous command failed?>

Comment (adarsh-zededa) [reply pkg/pillar/cmd/domainmgr/containerd.go]: > Also, in the silent case shouldn't you return nil?
This method returns nil in silent case at the end

>How do we know it is safe to drive on when a previous command failed?
There were cases where the clean-up was half done on device reboot (caused by this issue: https://www.pivotaltracker.com/story/show/171802157) because of which in some cases the container was deleted, rootfs inside bundle was unmounted but the bundle was not deleted. So the purpose I added silent mode was to delete/clean-up the things that are present and ignore the rest.
Clean-up in silent mode is only triggered in 2 cases
1) After we create a container if there is an error on some intermittent step while creating a bundle for the container. 
2) After device reboot, when we restart the AppInstance

Comment (adarsh-zededa) [reply pkg/pillar/cmd/domainmgr/containerd.go]: >Can avoid duplication in all these cases if you do

Done.

File: pkg/pillar/cmd/domainmgr/containerd.go (hunk 1 above)
Comment (adarsh-zededa) [reply pkg/pillar/cmd/domainmgr/containerd.go]: I checked on the device that after we called `container.Delete()` containerd's GC didn't clean-up the snapshot. 

>what happens if it is GC-ed and you try to delete it? Do we get an error?

After I deleted an app through eve, I waited for an hour to see if GC deletes the snapshot, it didn't

**Comment by rvs** [comment]
Hey @adarsh-zededa can you please rebase this on top of #749 ?

@eriknordmark -- please don't pull it in for now, ok?

**Comment by deitch** [comment]
@adarsh-zededa and I have been talking via slack. We are trying to get the GC part sorted. Either way, as long as the snapshot removal is idempotent, ie it can sanely handle it if the delete returns “no such snapshot” as distinct from “error deleting the snapshot”, then I see no issues with this.

**Comment by adarsh-zededa** [comment]
> Hey @adarsh-zededa can you please rebase this on top of #749 ?

Done. Had to do some minor changes.
1) Using the same UUID for Xen domain and container bundle.
2) There was an extra "]\n" in p9 of Xen config. Corrected that.

**Comment by adarsh-zededa** [comment]
EveImage check:  https://jenkins01.devt.zededa.net:8080/job/Smoketest-EveImageCheck/123/testReport/

**Comment by kalyan-nidumolu** [comment]
One good thing to note in the results is test_multiapps passed. But there
are 3 other failures. Can you check and confirm these are not related to
your code changes?

Thanks,
Kalyan

**Comment by adarsh-zededa** [comment]
@kalyan-nidumolu 
Error failures reason:
Error 1: App instance TestAppAssignedAudio-sc-supermicro-zc1-2020-03-16T16-57-29 to be online timed out
Reason: _qemu crashed - please restart application instance_.  
I'll be working on this next as part of  https://www.pivotaltracker.com/story/show/171826181

Error 2: Image 4.2.0-amd64 to become active on sc-supermicro-zc1 - CONDITION FAILED
Reason: _Reboot from agent vaultmgr, Reboot from agent nodeagent, Reboot from agent nim, Reboot from agent diag._ 
All the above failures are for the image: 4.2.0-amd64 on the device sc-supermicro-zc1

Error 3: Network Instance Create Failed body
Reason: _Subnet conflicts with network instance TestOneApp-colfax-sm300-gw5-NI-2020-03-16T20-34-07_
I guess this is not related to my changes. Need to look at what are to 2 conflicting apps and fix that.

**Comment by rvs** [comment]
I wanted to test it somewhat more today @kalyan-nidumolu -- hence the holdup with merging. We will merge this tomorrow (or should I say later today ;-))

**Comment by rvs** [comment]
Squashed and merged this @adarsh-zededa 

Thanks for the patch!
